src/PowerShellEditorServices/Analysis/AnalysisService.cs

```diff
@@ -590,6 +639,29 @@ public void Dispose()
         }
 
         #endregion
+
+        /// <summary>
+        /// Wraps the result of an execution of PowerShell to send back through
+        /// asynchronous calls.
+        /// </summary>
+        private class PowerShellResult
```

Comment:
I feel like this class could be put in some utility class rather than in this file - perhaps if `InvokePowerShellAsync` is used elsewhere.